cmd/recovereth/main.go

```diff
@@ -0,0 +1,396 @@
+func flipBit(j cryptoJSON, i uint) (cryptoJSON, error) {
+
+	if i/8 < uint(len(j.CipherText))/2 {
+		newText := flipBitInString(i, j.CipherText)
+		j.CipherText = newText
+		return j, nil
+	}
+	i -= uint(len(j.CipherText)) * 4
+
+	if i/8 < uint(len(j.CipherParams.IV))/2 {
+		newIV := flipBitInString(i, j.CipherParams.IV)
+		j.CipherParams.IV = newIV
+		return j, nil
+	}
+
+	i -= uint(len(j.CipherParams.IV)) * 4
+
+	// salt 32 bytes
+	salt, _ := j.KDFParams["salt"].(string)
+	if i/8 < uint(len(salt))/2 {
+		newSalt := flipBitInString(i, salt)
+		j.KDFParams["salt"] = newSalt
+		return j, nil
+	}
+	return j, fmt.Errorf("Exhausted")
+}
+
+func ensureInt(x interface{}) int {
+	res, ok := x.(int)
+	if !ok {
+		res = int(x.(float64))
+	}
+	return res
+}
+
+
+func getKDFKey(cryptoJSON cryptoJSON, auth string) ([]byte, error) {
+	authArray := []byte(auth)
+	salt, err := hex.DecodeString(cryptoJSON.KDFParams["salt"].(string))
+	if err != nil {
+		return nil, err
+	}
+	dkLen := ensureInt(cryptoJSON.KDFParams["dklen"])
+	if cryptoJSON.KDF == "scrypt" {
+		n := ensureInt(cryptoJSON.KDFParams["n"])
+		r := ensureInt(cryptoJSON.KDFParams["r"])
+		p := ensureInt(cryptoJSON.KDFParams["p"])
+		return scrypt.Key(authArray, salt, n, r, p, dkLen)
+
+	} else if cryptoJSON.KDF == "pbkdf2" {
+		c := ensureInt(cryptoJSON.KDFParams["c"])
+		prf := cryptoJSON.KDFParams["prf"].(string)
+		if prf != "hmac-sha256" {
+			return nil, fmt.Errorf("Unsupported PBKDF2 PRF: %s", prf)
+		}
+		key := pbkdf2.Key(authArray, salt, c, dkLen, sha256.New)
+		return key, nil
+	}
+	return nil, fmt.Errorf("Unsupported KDF: %s", cryptoJSON.KDF)
+}
+
+func aesCTRXOR(key, inText, iv []byte) ([]byte, error) {
+	aesBlock, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+	stream := cipher.NewCTR(aesBlock, iv)
+	outText := make([]byte, len(inText))
+	stream.XORKeyStream(outText, inText)
+	return outText, err
+}
+
+// decryptKeyV3x decrypts does not validate MAC during decryption
+func decryptKeyV3x(crypto cryptoJSON, auth string) (keyBytes []byte, err error) {
+	//keyId = uuid.Parse(keyProtected.Id)
+
+	iv, err := hex.DecodeString(crypto.CipherParams.IV)
+	if err != nil {
+		return nil, err
+	}
+
+	cipherText, err := hex.DecodeString(crypto.CipherText)
+	if err != nil {
+		return nil, err
+	}
+
+	derivedKey, err := getKDFKey(crypto, auth)
+	if err != nil {
+		return nil, err
+	}
+
+	plainText, err := aesCTRXOR(derivedKey[:16], cipherText, iv)
+	if err != nil {
+		return nil, err
+	}
+	return plainText, err
+}
+
+func testPasswordPrefixes(keyjson, auth string) error {
+	l := []string{"", "\n", "\r\n", " ", "\r"}
+	k := new(encryptedKeyJSONV3)
+	if err := json.Unmarshal([]byte(keyjson), k); err != nil {
+		log.Fatal(err)
+	}
+	addr := common.HexToAddress(k.Address)
+	for _, prefix := range l {
+		for _, suffix := range l {
+			pw := fmt.Sprintf("%v%v%v", prefix, auth, suffix)
+			keyBytes, err := decryptKeyV3x(k.Crypto, pw)
+			fmt.Printf("auth: %x\n", pw)
```

Comment:
I think it's better to use `fmt.Fprintf` instead of `fmt.Printf`. This will allow you to specify the file to output to, for example, `os.Stderr` or `os.Stdout'.